### ISSUE TITLE
Protect long term support census builds

### ISSUE THREAD

**Issue by atolopko-czi** [chanzuckerberg/cell-census#242]
## Motivation

Long Term Support (LTS) Census releases are intended to be maintained for 5 years.  It is critical that we do not accidentally delete these releases.

## Definition of Done

All objects of every LTS release should have AWS S3 "object lock" flag set.

## Tasks

- [ ] Update the builder's s3 upload operation to set the "object lock" flag on all uploaded S3 objects.

**Comment by atolopko-czi** [comment]
Superseded by #241

**Comment by atolopko-czi** [comment]
Reopening this story, since there a simple, additional level of protection we can add to LTS releases: S3 "object lock"

**Comment by atolopko-czi** [comment]
> Yeah, I'm gonna reach out to the AWS support to enable it. Shall we do this directly on the public bucket?

As opposed to testing on a temporary bucket? I think it's okay to do it on the public bucket, since we're not depending upon yet, so if it were to introduce any issues, we can fix.

**Comment by ebezzi** [comment]
> > Yeah, I'm gonna reach out to the AWS support to enable it. Shall we do this directly on the public bucket?
> 
> As opposed to testing on a temporary bucket? I think it's okay to do it on the public bucket, since we're not depending upon it yet, so object lock config were to introduce any issues, we can still fix.

No, as opposed to use the current bucket (the one in single-cell-prod).

**Comment by pablo-gar** [comment]
Please revisit after this is addressed: [Create Census data mirrors in the top 5 most used geographical locations](https://github.com/chanzuckerberg/cellxgene-census/issues/545)

**Comment by atolopko-czi** [comment]
With the upcoming 2nd LTS "emergency" census data release, @bkmartinjr realized an edge case whereby a prior LTS release could be deleted due to the 31 day deletion janitor logic, if it becomes unaliased. That will be addressed in another issue/PR. But more broadly, the concern over LTS data protection has been re-elevated. In light of that, this could be a good story to take on. Minimally, as a spike to determine whether this S3 object locking protection is easy to do. @ebezzi, I assigned to you provisionally, in case you have bandwidth. And @pablo-gar, for awareness or if you have concerns about allocating time to this.

**Comment by bkmartinjr** [comment]
I will drop a PR shortly to deal with the builder GC logic (adding a "do_no_delete" flag to release.json). 

Agree that it would also be nice to have an additional layer of protection described in this issue...

**Comment by ebezzi** [comment]
I'll resume looking into that.

**Comment by atolopko-czi** [comment]
When we implement Object Locking, we should also look into enabling ["Multi-factor authentication (MFA) delete"]( https://docs.aws.amazon.com/AmazonS3/latest/userguide/MultiFactorAuthenticationDelete.html?icmpid=docs_amazons3_console)
